storage/rocksdb/ha_rocksdb.h

```diff
@@ -1386,4 +1386,72 @@ struct Rdb_inplace_alter_ctx : public my_core::inplace_alter_handler_ctx {
   Rdb_inplace_alter_ctx(const Rdb_inplace_alter_ctx &);
   Rdb_inplace_alter_ctx &operator=(const Rdb_inplace_alter_ctx &);
 };
+
+/*
+  Helper class to control access/init to handlerton instance.
+  Contains a flag that is set if the handlerton is in an initialized, usable
+  state, plus a reader-write lock to protect it without serializing reads.
+  Since we don't have static initializers for the opaque mysql_rwlock type,
+  use constructor and destructor functions to create and destroy
+  the lock before and after main(), respectively.
+*/
+struct Rdb_hton_init_state {
+  struct Scoped_lock {
+    Scoped_lock(Rdb_hton_init_state& state, bool write) : m_state(state) {
+      if (write)
+        m_state.lock_write();
+      else
+        m_state.lock_read();
+    }
+    ~Scoped_lock() {
+      m_state.unlock();
+    }
+  private:
+    Rdb_hton_init_state& m_state;
+  };
+
```

Comment:
Please, also disable copy constructor and assign operator for Rdb_hton_init_state::Scoped_lock.

Reply:
Good catch, will prevent shenanigans and misuse later on if someone were to misunderstand the purpose.